test/e2e/lua/dynamic_configuration.go

```diff
@@ -210,6 +210,7 @@ var _ = framework.IngressNginxDescribe("Dynamic Configuration", func() {
 					return err
 				})
 			Expect(err).NotTo(HaveOccurred())
+			time.Sleep(5 * time.Second)
```

Comment:
Without this, the test is sometimes run with only 1 replica - since it doesn't wait for the 2nd replica to spin up. 

The only time it would work is when [this test](https://github.com/kubernetes/ingress-nginx/pull/2346/files#diff-8700c46a8379c603b5e41ccbaff6b71eR88) is run before this current test - since already 2 replicas are available at that point. Shout out to @zrdaley for helping troubleshoot this flaky test :) 

Should invest in putting together the ephemeral ingresses for each test - otherwise the e2e tests will being to get flaky very quickly.

Reply:
@andrewlouis93 maybe you can use something like [this](https://github.com/kubernetes/ingress-nginx/blob/ee46f486c77665096f366eb215aa135032291b5a/test/e2e/framework/framework.go#L273-L275) to wait for the replica count you expect

Reply:
Hmm interesting. 

So this sleep is actually happening right after an `UpdateDeployment` incantation: https://github.com/kubernetes/ingress-nginx/pull/2346/files#diff-8700c46a8379c603b5e41ccbaff6b71eR206

But because we are passing in that `updateFunc`, we are returning early before WaitForPodsReady is invoked. 

See: https://github.com/kubernetes/ingress-nginx/blob/c37ed7ecb62b7b9a2aa89f2320332acda60e9bba/test/e2e/framework/framework.go#L269

Reply:
@andrewlouis93 that should return early only if there is an error updating the deployment